src/handler/updateVrm.ts

```diff
@@ -1,69 +1,87 @@
       systemNumber,
       createdTimestamp,
     );
-    const validateVrmRes = validateVrm(currentRecord, newVrm);
-    if (validateVrmRes) {
-      return validateVrmRes;
+    const newVrmNotCorrectFormat = validateVrm(recipientRecord, newVrm);
+    if (newVrmNotCorrectFormat) {
+      return newVrmNotCorrectFormat;
     }
+    const userDetails = getUserDetails(event.headers.Authorization ?? '');
+    if (isCherishedTransfer === true) {
+      logger.debug('Performing cherished Transfer');
+      if (!thirdMark?.length) {
+        const newVrmExistsOnActiveRecord = await validateVrmExists(newVrm);
+        if (newVrmExistsOnActiveRecord) {
+          return newVrmExistsOnActiveRecord;
+        }
+      }
+      const [donorVehicleRecord, error] = await donorVehicle(newVrm, thirdMark) as [TechRecordType<'get'>, APIGatewayProxyResult];
+      if (error?.statusCode) {
+        return error;
+      }
+
+      const { recordsToArchive, recordsToUpdate } = processCherishedTransfer(
+        userDetails,
+        newVrm,
+        recipientRecord,
+        thirdMark,
+        donorVehicleRecord,
+      );
+      await updateVehicle(recordsToArchive, recordsToUpdate);
 
-    const techRecords = await searchByCriteria(SearchCriteria.PRIMARYVRM, newVrm);
-    logger.debug('Tech record search returned: ', techRecords);
-    const filteredVrm = techRecords.filter((x) => x.primaryVrm === newVrm && x.techRecord_statusCode !== StatusCode.ARCHIVED);
-    if (filteredVrm.length) {
       return addHttpHeaders({
-        statusCode: 400,
-        body: JSON.stringify(`Primary VRM ${newVrm} already exists`),
+        statusCode: 200,
+        body: JSON.stringify(recordsToUpdate[0]),
       });
     }
+    logger.debug('Correcting an error');
+    const newVrmExistsOnActiveRecord = await validateVrmExists(newVrm);
+    if (newVrmExistsOnActiveRecord) {
+      return newVrmExistsOnActiveRecord;
+    }
     logger.debug('identifier has been validated');
-    const userDetails = getUserDetails(event.headers.Authorization ?? '');
-    const [recordToArchive, newRecord] = processPatchVrmRequest(currentRecord, userDetails, newVrm, isCherishedTransfer);
+    const newRecipientRecord = processCorrectVrm(recipientRecord, userDetails, newVrm);
+
+    await correctVrm(newRecipientRecord);
 
-    if (isCherishedTransfer) {
-      await updateVehicle(
-        [recordToArchive],
-        [newRecord],
-      );
-    } else {
-      await correctVrm(newRecord);
-    }
     return addHttpHeaders({
       statusCode: 200,
-      body: JSON.stringify(formatTechRecord(newRecord)),
+      body: JSON.stringify(newRecipientRecord),
     });
-  } catch (error) {
-    console.log(error);
+  } catch (err) {
     return addHttpHeaders({
       statusCode: 500,
-      body: JSON.stringify(error),
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      body: JSON.stringify({ error: `Failed to update record : ${err}` }),
```

Comment:
could you use the `formatErrorMessage` function to return the error messsage (to keep error messages in a consitent format) ?